sound/soc/sof/ipc4-topology.c

```diff
@@ -1051,15 +1051,19 @@ static int sof_ipc4_init_audio_fmt(struct snd_sof_dev *sdev,
 	}
 
 	/* copy input format */
-	memcpy(&base_config->audio_fmt, &available_fmt->input_pin_fmts[i].audio_fmt,
-	       sizeof(struct sof_ipc4_audio_format));
+	if (available_fmt->input_pin_fmts)
+		memcpy(&base_config->audio_fmt, &available_fmt->input_pin_fmts[i].audio_fmt,
+		       sizeof(struct sof_ipc4_audio_format));
 
 	/* set base_cfg ibs/obs */
-	base_config->ibs = available_fmt->input_pin_fmts[i].buffer_size;
-	base_config->obs = available_fmt->output_pin_fmts[i].buffer_size;
+	if (available_fmt->input_pin_fmts)
+		base_config->ibs = available_fmt->input_pin_fmts[i].buffer_size;
+	if (available_fmt->output_pin_fmts)
+		base_config->obs = available_fmt->output_pin_fmts[i].buffer_size;
```

Comment:
I think I have asked this in the original PR, but the comment got lost @ranj063:
Let's say we have a module with 2 input and 1 output pin.
`sof_ipc4_init_audio_fmt()` is called with the `input_pin_fmts` and `num_input_formats`, we find the second pin_fmt as a match (i=1).
`output_pin_fmts` is allocated for one pin, no? We access data outside (output_pin_fmts[1])?

Reply:
We have 2 set of params: num_input_pins/num_output_pins and num_input_formats/num_output_formats. For WoV, there is no output pin. So it means, the num_output_pins and num_output_formats should be 0. In current WoV tplg, I set num_output_pins = 0 and num_output_formats = 1. This can work without this PR. (num_output_formats = 1)

However, I still think we need this patch for the sanity check in case someone set the num_input_formats or num_output_formats to 0 in tplg, which will cause kernel panic. Kernel should handle such exception.